fdbserver/KeyValueStoreShardedRocksDB.actor.cpp

```diff
@@ -2028,13 +2071,33 @@ struct ShardedRocksDBKeyValueStore : IKeyValueStore {
 				}
 			}
 
+			// Check for number of range deletes in shards.
+			// TODO: Disable this once RocksDB is upgraded to a version with range delete improvement.
+			if (SERVER_KNOBS->ROCKSDB_CF_RANGE_DELETION_LIMIT) {
+				rocksdb::FlushOptions fOptions;
+				fOptions.wait = true;
+				fOptions.allow_write_stall = true;
+
+				for (auto shard : (*a.dirtyShards)) {
```

Comment:
Will this be expensive? Will it up the latency on commits being done inline w/ the commit?

Reply:
I would expect latency to go up. As we do commit every 5 sec, it shouldn't affect other parts of the system. I added a knob to control blocking/non-blocking flush. We could do non-blocking flush if latency is an issue.